test/e2e/extensions/high_availability_test.go

```diff
@@ -64,9 +64,38 @@ func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
 	assert.NoError(t, framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{})) // nolint: errcheck
 }
 
+func TestGameServerCreationRightAfterDeletingOneExtensionsPod(t *testing.T) {
+	logger := e2eframework.TestLogger(t)
+	ctx := context.Background()
+
+	if !runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions) {
+		t.Skip("Skip test. SplitControllerAndExtensions feature is not enabled")
+	}
+
+	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+
+	list, err := getAgoneseExtensionsPods(ctx)
+	logger.Infof("Length of pod list is %v", len(list.Items))
+	for i := range list.Items {
+		logger.Infof("Name of extensions pod %v: %v", i, list.Items[i].ObjectMeta.Name)
+	}
+	require.NoError(t, err, "Could not get list of Extension pods")
+
+	logger.Infof("Removing one of the Extensions Pods: %v", list.Items[1].ObjectMeta.Name)
+	deleteAgonesExtensionsPod(ctx, t, false)
+
+	gs := framework.DefaultGameServer(defaultNs)
+	logger.Infof("Creating game-server %s...", gs.Name)
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, defaultNs, gs)
```

Comment:
so here's a thought.. the extensions pod is an API pod. For the most part, it's not exactly involved in the "wait until ready" part and doesn't have a lot of interaction with the `GameServer` after creation. Furthermore, this only really does the one interaction, and does it right after we delete the extensions pod, when the pod should be lame-ducked. The more meaningful time for it to try is right after the readiness failure and the connection is drained.

We can't detect that timing exactly, of course, but I would proposed that a meaningful test here is just to loop for like 30s, creating and delete a gameserver. That probes the API continuously for that period and should give us a more meaningful signal.

Reply:
I added a loop around the create and delete step. Please take a look.